audb/core/load.py

```diff
@@ -204,6 +207,76 @@ def _files_duration(
     db._files_duration = durs.to_dict()
 
 
+def _get_attachments_from_cache(
+        attachments: typing.Sequence[str],
+        db_root: str,
+        deps: Dependencies,
+        cached_versions: CachedVersions,
+        flavor: Flavor,
+        num_workers: int,
+        verbose: bool,
+) -> typing.Sequence[str]:
+    r"""Copy files from cache.
+
+    This function copies all files
+    associated with the requested attachemnts
+    from other cached versions
+    to the new database folder.
+
+    Args:
+        attachments: sequence of attachment IDs
+        db_root: database root
+        deps: dependency object
+        cached_versions: object containing information
+            on existing cached versions of the database
+        flavor: database flavor object
+        num_workers: number of workers to use
+        verbose: if ``True`` show progress bar
+
+    Returns:
+        list of attachment IDs that couldn't be found in cache
+
+    """
+    db_root_cached = [x[1] for x in cached_versions]
+
+    paths = [
+        deps._df[deps._df['archive'] == attachment].index[0]
+        for attachment in attachments
+    ]
```

Comment:
If we pass `db` to the function I think we could replace with:

`paths = [db.attachments[attachment].path for attachment in attachments]`

Reply:
My first idea was to not add another argument to `_get_attachments_from_cache()`, but your solution is easier to understand, so I changed it accordingly.